src/ts/JsonEditorHelper.class.ts

```diff
@@ -121,14 +122,23 @@ export default class JsonEditorHelper {
   // save a file or metadata depending on what was loaded
   save(): void {
     if (this.editorDiv.dataset.what === 'file') {
-      return this.saveFile();
-    } else if (this.editorDiv.dataset.what === 'metadata') {
+      return this.saveFile(false);
+    }
+
+    if (this.editorDiv.dataset.what === 'metadata') {
       return this.saveMetadata();
     }
+
+    // toggle save menu so user can select what to save: file or metadata
+    if (this.editorDiv.dataset.what === '') {
+      if ($('#json-save-dropdown').next('div').is(':hidden')) {
+        $('#json-save-dropdown').dropdown('toggle');
+      }
+    }
   }
 
-  saveFile(): void {
-    if (typeof this.currentUploadId === 'undefined') {
+  saveFile(newFile = true): void {
```

Comment:
maybe instead of having a boolean flag we can have saveFile and saveNewFile? It'll convey more meaning.

Reply:
Good idea. I will rework it.